rxbinding/src/main/java/com/jakewharton/rxbinding/view/ObservableGestureListener.java

```diff
@@ -0,0 +1,110 @@
+package com.jakewharton.rxbinding.view;
+
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+
+import rx.Observable;
+import rx.subjects.PublishSubject;
+
+final class ObservableGestureListener implements GestureDetector.OnGestureListener {
+
+    final View view;
+
+    private PublishSubject<ViewGestureEvent> downGestureObservable;
+    private PublishSubject<ViewGestureEvent> showPressGestureObservable;
+    private PublishSubject<ViewGestureEvent> singleTapUpGestureObservable;
+    private PublishSubject<ViewGestureScrollEvent> scrollGestureObservable;
+    private PublishSubject<ViewGestureEvent> longPressGestureObservable;
+    private PublishSubject<ViewGestureFlingEvent> flingGestureObservable;
```

Comment:
Not at the computer, but IIRC the issue with the custom OnSubscribe is
setting the OnTouchListener. Since we can only have one at a time, we
needed to be able to add these without requiring that this be the only
listener, so I chose this pattern so that you could get the onTouches
subscription and simply pass it to this, in addition to doing whatever
other things you want with it. Let me know if I'm wrong there, haven't
looked in a while.

Also, I like the single PublishSubject with ofType, I didn't know about
that.

On Sat, Sep 26, 2015, 11:05 PM Jake Wharton <notifications@github.com>
wrote:

> In
> rxbinding/src/main/java/com/jakewharton/rxbinding/view/ObservableGestureListener.java
> <https://github.com/JakeWharton/RxBinding/pull/128#discussion_r40501187>:
>
> > +import android.view.MotionEvent;
> > +import android.view.View;
> > +
> > +import rx.Observable;
> > +import rx.subjects.PublishSubject;
> > +
> > +final class ObservableGestureListener implements GestureDetector.OnGestureListener {
> > +
> > +    final View view;
> > +
> > +    private PublishSubject<ViewGestureEvent> downGestureObservable;
> > +    private PublishSubject<ViewGestureEvent> showPressGestureObservable;
> > +    private PublishSubject<ViewGestureEvent> singleTapUpGestureObservable;
> > +    private PublishSubject<ViewGestureScrollEvent> scrollGestureObservable;
> > +    private PublishSubject<ViewGestureEvent> longPressGestureObservable;
> > +    private PublishSubject<ViewGestureFlingEvent> flingGestureObservable;
>
> Actually we'd probably create an OnSubscribe class like the other
> listener bindings and just emit ViewGestureEvent. Then callers can filter
> or we can provide filters.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JakeWharton/RxBinding/pull/128/files#r40501187>.
>